File: src/test_polars/utils/process_csv.py
"""Module contains functions for reading CSV files into Polars DataFrames."""

from pathlib import Path

import polars as pl


def get_csv(file_path: Path) -> pl.DataFrame:
    """
    Read a CSV file into a Polars DataFrame.

    Parameters
    ----------
    file_path: Path
        The path to the CSV file.

    Returns
    -------
    polars.DataFrame
        The Polars DataFrame.

    """
    return pl.read_csv(file_path)

File: src/utils/process_csv.rs
use std::path::PathBuf;

use polars::{
    io::SerReader,
    prelude::{CsvReadOptions, DataFrame},
};

/// Reads a CSV file into a Polars DataFrame.
///
/// # Errors
///
/// If there is an error while reading the file, an empty DataFrame is returned.
pub fn get_csv(path: PathBuf) -> DataFrame {
    match CsvReadOptions::default()
        .with_has_header(true)
        .try_into_reader_with_file_path(Some(path))
    {
        Ok(csv_reader) => csv_reader.finish().unwrap_or(DataFrame::empty()),
        Err(_) => DataFrame::empty(),
    }
}
